app/src/main/java/com/framgia/wsm/widget/holidaycalendar/HolidayCalendarView.java

```diff
@@ -0,0 +1,368 @@
+        mMonthTitlePaint.setFakeBoldText(true);
+        mMonthTitlePaint.setAntiAlias(true);
+        mMonthTitlePaint.setTextSize(mMonthLabelTextSize);
+        mMonthTitlePaint.setTypeface(Typeface.create(monthTitleTypeface, Typeface.BOLD));
+        mMonthTitlePaint.setColor(mMonthTextColor);
+        mMonthTitlePaint.setTextAlign(Paint.Align.CENTER);
+        mMonthTitlePaint.setStyle(Paint.Style.FILL);
+
+        mReservationInquiryStatusPaint = new Paint();
+        mReservationInquiryStatusPaint.setFakeBoldText(true);
+        mReservationInquiryStatusPaint.setAntiAlias(true);
+        mReservationInquiryStatusPaint.setTextAlign(Paint.Align.CENTER);
+        mReservationInquiryStatusPaint.setStyle(Paint.Style.FILL);
+        mReservationInquiryStatusPaint.setAlpha(SELECTED_CIRCLE_ALPHA);
+
+        mSelectedCirclePaint = new Paint();
+        mSelectedCirclePaint.setColor(Color.WHITE);
+        mSelectedCirclePaint.setFakeBoldText(true);
+        mSelectedCirclePaint.setAntiAlias(true);
+        mSelectedCirclePaint.setTextAlign(Paint.Align.CENTER);
+        mSelectedCirclePaint.setStyle(Paint.Style.FILL);
+
+        mMonthDayLabelPaint = new Paint();
+        mMonthDayLabelPaint.setAntiAlias(true);
+        mMonthDayLabelPaint.setTextSize(mMonthDayLabelTextSize);
+        mMonthDayLabelPaint.setColor(mDayTextColor);
+        mMonthDayLabelPaint.setTypeface(Typeface.create(monthTitleTypeface, Typeface.NORMAL));
+        mMonthDayLabelPaint.setStyle(Paint.Style.FILL);
+        mMonthDayLabelPaint.setTextAlign(Paint.Align.CENTER);
+        mMonthDayLabelPaint.setFakeBoldText(true);
+
+        mMonthNumPaint = new Paint();
+        mMonthNumPaint.setAntiAlias(true);
+        mMonthNumPaint.setTextSize(mMiniDayNumberTextSize);
+        mMonthNumPaint.setStyle(Paint.Style.FILL);
+        mMonthNumPaint.setTextAlign(Paint.Align.CENTER);
+        mMonthNumPaint.setFakeBoldText(false);
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        drawMonthTitle(canvas);
+        drawMonthDayLabels(canvas);
+        drawMonthNums(canvas);
+    }
+
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        setMeasuredDimension(MeasureSpec.getSize(widthMeasureSpec),
+                mRowHeight * mNumRows + mMonthHeaderSize + mMonthHeaderSize / 8);
+    }
+
+    @Override
+    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
+        mWidth = w;
+    }
+
+    @Override
+    public boolean onTouchEvent(MotionEvent event) {
+        if (event.getAction() == MotionEvent.ACTION_UP) {
+            performClick();
+            Date date = getDayFromLocation(event.getX(), event.getY());
+            if (date == null) {
+                return false;
+            }
+            for (HolidayCalendarDate holidayCalendarDate : mHolidayCalendarDates) {
+                if (date.equals(holidayCalendarDate.getDate())) {
+                    if (mOnDayClickListener != null) {
```

Comment:
WARNING: [CollapsibleIfStatements] These nested if statements could be combined
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/basic.html#CollapsibleIfStatements